src/main/java/uk/gov/hmcts/juror/api/moj/report/standard/IncompleteServiceReport.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.juror.api.moj.report.standard;
+
+import com.querydsl.core.Tuple;
+import com.querydsl.jpa.impl.JPAQuery;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.juror.api.moj.controller.reports.request.StandardReportRequest;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.AbstractReportResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.StandardReportResponse;
+import uk.gov.hmcts.juror.api.moj.domain.QJuror;
+import uk.gov.hmcts.juror.api.moj.domain.QJurorPool;
+import uk.gov.hmcts.juror.api.moj.report.AbstractStandardReport;
+import uk.gov.hmcts.juror.api.moj.report.DataType;
+import uk.gov.hmcts.juror.api.moj.repository.CourtLocationRepository;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@SuppressWarnings("PMD.LawOfDemeter")
+public class IncompleteServiceReport extends AbstractStandardReport {
+
+    private final CourtLocationRepository courtLocationRepository;
+
+    @Autowired
+    public IncompleteServiceReport(CourtLocationRepository courtLocationRepository) {
+        super(
+            QJuror.juror,
+            DataType.JUROR_NUMBER,
+            DataType.FIRST_NAME,
+            DataType.LAST_NAME,
+            DataType.POOL_NUMBER_BY_JP,
+            DataType.NEXT_ATTENDANCE_DATE);
+        this.courtLocationRepository = courtLocationRepository;
+        isCourtUserOnly();
+        addAuthenticationConsumer(request -> checkOwnership(request.getLocCode(), false));
+    }
+
+
+    @Override
+    protected void preProcessQuery(JPAQuery<Tuple> query, StandardReportRequest request) {
+        query
+            .where(QJurorPool.jurorPool.nextDate.loe(request.getDate()))
+            .where(QJurorPool.jurorPool.owner.eq(request.getLocCode()))
+            .where(QJurorPool.jurorPool.status.status.in(List.of(2, 3, 4)))
+            .orderBy(QJuror.juror.jurorNumber.asc())
+            .fetch();
```

Comment:
Fetch not needed apart of the pre-process

Reply:
removed